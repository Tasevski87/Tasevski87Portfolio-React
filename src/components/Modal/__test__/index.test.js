// __tests__/Modal.test.js with hard coded categories
import React from 'react';
import { render, cleanup, fireEvent } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import Modal from '..';

const mockToggleModal = jest.fn();
const currentPhoto = {
    name: 'Social Network Api',
    category: 'projects',
    description: <a href="https://github.com/Tasevski87/Social-Network-API"> NoSQL Social Network API</a>,
    index: 1
};

afterEach(cleanup);

describe('Modal component', () => {
    it('renders', () => {
        render(<Modal
            onClose={mockToggleModal}
            currentPhoto={currentPhoto}
        />);
    })

    it('matches snapshot DOM node structure', () => {
        const { asFragment } = render(<Modal
            onClose={mockToggleModal}
            currentPhoto={currentPhoto}
        />);

        expect(asFragment()).toMatchSnapshot();
    });
})

describe('Click Event', () => {
    it('calls onClose handler', () => {
        const { getByText } = render(<Modal
            onClose={mockToggleModal}
            currentPhoto={currentPhoto}
        />);
        fireEvent.click(getByText('Close'))

        expect(mockToggleModal).toHaveBeenCalledTimes(1);
    });
})  