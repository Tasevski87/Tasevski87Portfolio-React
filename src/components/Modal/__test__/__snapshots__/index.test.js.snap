// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal component matches snapshot DOM node structure 1`] = `
<DocumentFragment>
  <div
    class="modalBackdrop"
  >
    <div
      class="modalContainer"
    >
      <h3
        class="modalTitle"
      >
        Social Network Api 
      </h3>
      <img
        alt="current category"
        src="1.jpg"
      />
      <p>
        <a
          href="https://github.com/Tasevski87/Social-Network-API"
        >
           NoSQL Social Network API
        </a>
      </p>
      <button
        type="button"
      >
        Close
      </button>
    </div>
  </div>
</DocumentFragment>
`;
